frontends/web/src/utils/walletconnect-eth-sign-handlers.ts

```diff
@@ -0,0 +1,195 @@
+  if (!accountDetail.success) {
+    console.log('Failed in fetching account name and code'); //silently fails
+    return { accountName: '', accountCode: '' };
+  }
+  const { code, name } = accountDetail;
+  return { accountName: name, accountCode: code };
+};
+
+export async function handleWcEthSignRequest(method: string, args: TEthSignHandlerParams) {
+  switch (method) {
+  case EIP155_SIGNING_METHODS.ETH_SIGN:
+  case EIP155_SIGNING_METHODS.PERSONAL_SIGN:
+    await ethSignHandler(args, method);
+    break;
+  case EIP155_SIGNING_METHODS.ETH_SIGN_TYPED_DATA:
+  case EIP155_SIGNING_METHODS.ETH_SIGN_TYPED_DATA_V3:
+  case EIP155_SIGNING_METHODS.ETH_SIGN_TYPED_DATA_V4:
+    await ethSignTypedDataHandler(args);
+    break;
+  case EIP155_SIGNING_METHODS.ETH_SIGN_TRANSACTION:
+  case EIP155_SIGNING_METHODS.ETH_SEND_TRANSACTION:
+    await ethSignOrSendTransactionHandler(args, method);
+    break;
+  default:
+    console.log(`${method} is unsupported`);
+  }
+}
+
+/**
+     * Wallet Connect's ETH_SIGN gives the params as [address, message]
+     * while PERSONAL_SIGN gives them as [message, address]
+*/
+
+async function ethSignHandler({ params, launchSignDialog, topic, id, currentSession }: TEthSignHandlerParams, method: string) {
+  const isPersonalSign = method === EIP155_SIGNING_METHODS.PERSONAL_SIGN;
+  const requestParams = params.request.params;
+  const accountAddress = isPersonalSign ? requestParams[1] : requestParams[0];
+  const signingData = isPersonalSign ? requestParams[0] : requestParams[1];
+  const decoded = decodeEthMessage(signingData);
+  if (decoded === null) {
+    alertUser(t('walletConnect.signingRequest.decodeError'));
+    return;
+  }
+  const { accountName, accountCode } = await fetchAccountNameAndAddress(accountAddress);
+  const apiCaller = async () => {
+    const result = await ethSignMessage(accountCode, signingData);
+    if (!result.success) {
+      return { success: false, error: result };
+    }
+    return {
+      response: { id, jsonrpc: '2.0', result: result.signature },
+      success: true
+    };
+  };
+
+  launchSignDialog({
+    topic,
+    id,
+    apiCaller,
+    dialogContent: {
+      signingData: decoded,
+      currentSession,
+      accountName,
+      accountAddress,
+      chain: params.chainId,
+      method: t('walletConnect.signingRequest.method.signMessage')
+    }
+  });
+}
+
+async function ethSignTypedDataHandler({ params, launchSignDialog, topic, id, currentSession }: TEthSignHandlerParams) {
+  const requestParams = params.request.params;
+  const accountAddress = requestParams[0];
+  const data = requestParams[1];
+  let typedData: any;
+  const { accountName, accountCode } = await fetchAccountNameAndAddress(accountAddress);
+
+  try {
+    typedData = JSON.parse(data);
+  } catch (e) {
+    console.error('Failed to parse JSON', e);
+    typedData = undefined;
+    // If JSON parsing fails, typedData will be undefined.
```

Comment:
thanks! 

one alternative could be to just assign the string to this variable `typedData = data` and then it signs that string, but maybe this never happens.

cc @Tomasvrba WDYT?

Reply:
Maybe do it like I did with the hex data decoding here: https://github.com/digitalbitbox/bitbox-wallet-app/blob/5ccc03aae817aea0fde54199bf1dbb91f208eecc/frontends/web/src/utils/walletconnect-eth-sign-handlers.ts#L80-L82

ie. return when the parsing/decoding fails? There's probably a reason why it failed so we shouldn't send it to the BB to sign. Or if it was just a glitch in the app, the user can try again, but better be safe imo